pkg/daemon/update.go

```diff
@@ -1188,7 +1194,8 @@ func (dn *Daemon) deleteStaleData(oldIgnConfig, newIgnConfig ign3types.Config) e
 		}
 		if _, err := os.Stat(noOrigFileStampName(f.Path)); err == nil {
 			if err := os.Remove(noOrigFileStampName(f.Path)); err != nil {
-				return errors.Wrapf(err, "deleting noorig file stamp %q: %v", noOrigFileStampName(f.Path), err)
+				errs := kubeErrs.NewAggregate([]error{err, err})
```

Comment:
Even though there are two `err` variables here, in the case where you're aggregating both of them, one of them will shadow the other. The error that matters in this case is the one returned by `os.Remove` since the error from `os.Stat` will always be nil. I suggest something like this instead:

```golang
if _, err := os.Stat(noOrigFileStampName(f.Path)); err == nil {
  if delErr := os.Remove(noOrigFileStampName(f.Path)); delErr != nil {
    return fmt.Errorf("deleting noorig file stamp %q: %w", noOrigFileStampName(f.Path), delErr)
  }
}
```

This way, it's obvious that the delete error is what is being returned.